nav2_rviz_plugins/src/selector.cpp

```diff
@@ -0,0 +1,184 @@
+  pub_smoother_ = client_node_->create_publisher<std_msgs::msg::String>("smoother_selector", qos);
+
+  main_layout_ = new QHBoxLayout;
+  left_layout_ = new QVBoxLayout;
+  right_layout_ = new QVBoxLayout;
+  controller_ = new QComboBox;
+  planner_ = new QComboBox;
+  goal_checker_ = new QComboBox;
+  smoother_ = new QComboBox;
+
+  main_layout_->setContentsMargins(10, 10, 10, 10);
+
+  left_layout_->addWidget(new QLabel("Controller"));
+  left_layout_->addWidget(controller_);
+  left_layout_->addWidget(new QLabel("Planner"));
+  left_layout_->addWidget(planner_);
+  right_layout_->addWidget(new QLabel("Goal Checker"));
+  right_layout_->addWidget(goal_checker_);
+  right_layout_->addWidget(new QLabel("Smoother"));
+  right_layout_->addWidget(smoother_);
+
+  main_layout_->addLayout(left_layout_);
+  main_layout_->addLayout(right_layout_);
+
+  setLayout(main_layout_);
+  timer_.start(200, this);
+
+  connect(
+    controller_, QOverload<int>::of(&QComboBox::activated), this,
+    &Selector::setController);
+
+  connect(
+    planner_, QOverload<int>::of(&QComboBox::activated), this,
+    &Selector::setPlanner);
+
+  connect(
+    goal_checker_, QOverload<int>::of(&QComboBox::activated), this,
+    &Selector::setGoalChecker);
+
+  connect(
+    smoother_, QOverload<int>::of(&QComboBox::activated), this,
+    &Selector::setSmoother);
+}
+
+Selector::~Selector()
+{
+}
+
+// Publish the selected controller or planner
+void Selector::setSelection(
+  QComboBox * combo_box, rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher)
+{
+  // If "default" option is selected, it gets removed and the next item is selected
+  if (combo_box->findText("Default") != -1) {
+    combo_box->removeItem(0);
+  }
+
+  // if there are no plugins available, return
+  if (combo_box->count() == 0) {
+    return;
+  }
+
+  std_msgs::msg::String msg;
+  msg.data = combo_box->currentText().toStdString();
+
+  publisher->publish(msg);
+  timer_.start(200, this);
+}
+
+// Call setSelection() for controller
+void Selector::setController()
+{
+  setSelection(controller_, pub_controller_);
+}
+
+// Call setSelection() for planner
+void Selector::setPlanner()
+{
+  setSelection(planner_, pub_planner_);
+}
+
+// Call setSelection() for goal checker
+void Selector::setGoalChecker()
+{
+  setSelection(goal_checker_, pub_goal_checker_);
+}
+
+// Call setSelection() for smoother
+void Selector::setSmoother()
+{
+  setSelection(smoother_, pub_smoother_);
+}
+
+// Load the available plugins into the combo box
+void Selector::pluginLoader(
+  rclcpp::Node::SharedPtr node,
+  const std::string & server_name,
+  const std::string & plugin_type,
+  QComboBox * combo_box)
+{
+  auto parameter_client = std::make_shared<rclcpp::SyncParametersClient>(node, server_name);
+
+  // Wait for the service to be available before calling it
+  bool server_unavailable = false;
+  while (!parameter_client->wait_for_service(3s)) {
```

Comment:
We get the issue exactly here. As you see indeed have a logger for it too below. It's always the `controller_server` which is not available.. Probably we could wait couple of more seconds here?

Reply:
Possibly yes! Or you could wait once, then just pass the callback and come back again in another 200ms to try again, rather than `while` looping